include/hex.hpp

```diff
@@ -0,0 +1,169 @@
+#pragma once
+#include <array>
+#include <vector>
+#include <concepts>
+#include <utility>
+#include <optional>
+#include <raymath.h>
+
+/*
+Some notes about what the code means.
+Most taken from https://www.redblobgames.com/grids/hexagons/#coordinates
+Please, read that, especially the section on Cube and Axial coordinates
+This header defines some user defined literals operators
+They are: _LU (Left Up), _L (Left), _LD (Left Down), _RU (Right Up), _R (Right), _RD (Right Down)
+To get a cell, you can take an existing one, and add numbers suffixed by those UDLs
+So, for example:
+    HexCoords two_to_the_right = current + 2_R;
+*/
+
+struct HexCoords;
+struct EdgeCoords;
+
+// Operators to get offsets from a cell. They are Left and Right combined with Up, Down, or just
+HexCoords operator "" _LU (size_t x);
```

Comment:
na systemie macos size_t odmawia tu posłuszeństwa z błędem:
`error: no matching literal operator for call to 'operator""_R' with argument of type 'unsigned long long' or 'const char *', and no matching literal operator template`
Czy unsigned long long zadziała crossplatformowo?